File: Seccion 17 - Archivos o Ficheros en C++/04_ejercicio_02.cpp
/*
    02. Lectura de un archivo de texto preguntando el nombre o la 
    ubicación.
*/

#include <iostream>
#include <stdlib.h>
#include <fstream>
#include <string.h>

using namespace std;

void lectura();

int main() {

    lectura();

    system("pause");
    return 0;
}

void lectura() {
    ifstream archivo;
    string nombreArchivo, texto;

    cout << "\nDigite el nombre o la ubicación del archivo: ";
    getline(cin, nombreArchivo);

    archivo.open(nombreArchivo.c_str(), ios::in); // Abrimos el archivo en modo lectura

    if (archivo.fail()) {
        cout << "\nNo se pudo abrir el archivo." << endl;
        exit(1);
    }

    while (!archivo.eof()) { // Mientras no sea el final del archivo
        getline(archivo, texto);
        cout << texto << endl;
    }

    archivo.close();
}

File: Seccion 17 - Archivos o Ficheros en C++/Seccion_17_Python/04_ejercicio_02.py

def lectura():
    nombreArchivo: str
    texto: str
    
    nombreArchivo = input("\nDigite el nombre o la ubicación del archivo: ")
    
    try:
        archivo = open(nombreArchivo, "r")
    except FileNotFoundError:
        print("\nNo se pudo abrir el archivo.")
    
    texto = archivo.readlines()
    
    for linea in texto:
        print(linea, end="")
    
    print()

def main():
    
    lectura()

if __name__ == "__main__":
    main()
    
    